validation/src/main/scala/hmda/validation/rules/lar/macro/Q007.scala

```diff
@@ -0,0 +1,37 @@
+package hmda.validation.rules.lar.`macro`
+
+import akka.actor.ActorSystem
+import akka.stream.ActorMaterializer
+import com.typesafe.config.ConfigFactory
+import hmda.model.fi.lar.LoanApplicationRegister
+import hmda.validation.dsl.Result
+import hmda.validation.rules.AggregateEditCheck
+import hmda.validation.dsl.PredicateCommon._
+import hmda.validation.dsl.PredicateSyntax._
+import hmda.validation.rules.lar.`macro`.MacroEditTypes._
+
+import scala.concurrent.{ ExecutionContext, Future }
+
+object Q007 extends AggregateEditCheck[LoanApplicationRegisterSource, LoanApplicationRegister] {
+
+  val config = ConfigFactory.load()
+  val multiplier = config.getInt("hmda.validation.macro.Q007.numOfLarsMultiplier")
+
+  override def name = "Q007"
+
+  override def apply(lars: LoanApplicationRegisterSource)(implicit system: ActorSystem, materializer: ActorMaterializer, ec: ExecutionContext): Future[Result] = {
+
+    val approvedButNotAccepted =
+      count(lars.filter(lar => lar.actionTakenType == 2))
+
+    val total = count(lars)
```

Comment:
Do most Macro edits require calculating against the total number of lars in the file? If so, could it make sense to calculate the total number just once and send the value to each Macro edit as a parameter?